src/mpi/datatype/typerep/dataloop/dataloop_create_struct.c

```diff
@@ -304,6 +304,9 @@ static int create_flattened_struct(MPI_Aint count,
              * add to flattened representation */
             if (sz > 0) {
                 segp = MPIR_Segment_alloc(NULL, (MPI_Aint) blklens[i], oldtypes[i]);
+                if (NULL == segp) {
+                    return create_struct_memory_error();
```

Comment:
This is weird. All it does is return `-1`, can't we use a macro for it?

Reply:
Because this is consistent with the other usage in this file. It doesn't use the typical `mpi_errno` error handling throughout the rest of the code.

Reply:
I agree this is just consistent with the rest of the file, but can we use `if (!segp)` instead?

Reply:
Done